types/apple-signin-api/index.d.ts

```diff
@@ -31,17 +35,18 @@ declare namespace AppleSignInAPI {
     }
     // https://developer.apple.com/documentation/signinwithapplejs/authi
     interface AuthI {
-        init: (config: ClientConfigI) => Promise<void>;
-        signIn: (signInConfig?: ClientConfigI) => Promise<SignInResponseI | SignInErrorI>;
+        init: (config: ClientConfigI) => void;
+        signIn: (signInConfig?: ClientConfigI) => Promise<SignInResponseI>;
         renderButton: () => void;
     }
     // https://developer.apple.com/documentation/signinwithapplejs/clientconfigi
     interface ClientConfigI {
-        clientId: string;
-        redirectURI: string;
-        scope: string;
-        state: string;
-        usePopup: boolean;
+        clientId?: string;
+        redirectURI?: string;
+        scope?: string;
+        state?: string;
+        nonce?: string;
+        usePopup?: boolean;
```

Comment:
All these options are optional, because they can also be specified in meta tags: https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple#3235722